pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -3060,7 +3060,21 @@ public void onUpdate(TopicPolicies policies) {
                 checkReplicationAndRetryOnFailure();
             }
 
-            checkDeduplicationStatus();
+            /**
+             * When execute command "pulsar-admin topicPolicies get-deduplication persistent://{tenant}/{ns}/{topic}"
+             * but "store.asyncRemoveCursor('pulsar.dedup')" failure, the 'pulsar-dedup' cursor will be disabled,
+             * zk-node: "/managed-ledgers/{tenant}/{ns}/persistent/{topic}/pulsar.dedup" still exists.
+             * Then this topic's ledger-store will not be deleted anymore, even if all message has been acknowledged.
+             * So when `checkDeduplicationStatus()` failure, stop this topic.
+             */
+            try {
+                checkDeduplicationStatus().get();
```

Comment:
At first glance, we cannot call sync method inside async method.

Reply:
Why ? I tried it , it works.